lib/devise/multi_email/models/authenticatable.rb

```diff
@@ -48,22 +48,33 @@ def email=(new_email)
       module ClassMethods
         def find_first_by_auth_conditions(tainted_conditions, opts = {})
           filtered_conditions = devise_parameter_filter.filter(tainted_conditions.dup)
-          email = filtered_conditions.delete(:email)
+          criteria = filtered_conditions.extract!(:email, :unconfirmed_email)
```

Comment:
@allenwq the `:unconfirmed_email` option exists in the scenario where there's an unconfirmed email:

https://github.com/plataformatec/devise/blob/master/lib/devise/models/confirmable.rb#L363

Overall, this is a potential because the intent with the refactor is to let this gem work seamlessly with all default functionality in Devise when specifying a new email address (which is "unconfirmed" by default). The intent is to allow Devise to use it's built-in ability to confirm "unconfirmed" emails.